### PR TITLE
Fixes error reporting for unsupported layout

### PR DESCRIPTION
**Pull request by aghassemi**
Fixes errors such as `undefined#undefined Layout not supported: responsive​​​` to `AMP-CAROUSEL#AMP_15  Layout not supported: responsive`

Also `layout not supported` should be a user error.

### CODE REVIEW

File: src/custom-element.js [hunk 1]
```diff
@@ -849,7 +849,8 @@ function createBaseCustomElementClass(win) {
           this.layout_ = applyLayout_(this);
           if (this.layout_ != Layout.NODISPLAY &&
```
Comment (jridgewell) [inline src/custom-element.js]: I'm not sure this should be in here now that we can delay the upgrading of Elements. And, it's just duplication if the upgrade is sync, right? /cc @dvoytenko

File: src/custom-element.js [hunk 2]
```diff
@@ -849,7 +849,8 @@ function createBaseCustomElementClass(win) {
...
             !this.implementation_.isLayoutSupported(this.layout_)) {
-            throw new Error('Layout not supported for: ' + this.layout_);
+            throw user().createError('Layout not supported for: ' +
```
Comment (jridgewell) [inline src/custom-element.js]: This wording seems off.

File: src/custom-element.js (hunk 1 above)
Comment (aghassemi) [inline src/custom-element.js]: Let's have a different PR/Issue if we need to clean up `connectedCallback`. I am not exactly sure what needs go away here.

Comment (jridgewell) [inline src/custom-element.js]: https://github.com/ampproject/amphtml/issues/5980

Comment (dvoytenko) [inline src/custom-element.js]: This clearly looks like a dupe.

**Pull request merged**